sdk/tools/kbdtool/parser.c

```diff
@@ -146,8 +146,8 @@ getVKNum(IN PCHAR p)
     if ((*p == '0') && ((*(p + 1) == 'x') || (*(p + 1) == 'X')))
     {
         /* Get the key number from the hex string */
-        *(p + 1) = 'x';
-        if (sscanf(p, "0x%x", &KeyNumber) == 1) return KeyNumber;
+        if (sscanf(p + 2, "%x", &KeyNumber) == 1 && KeyNumber <= 0xFF)
+            return KeyNumber;
```

Comment:
Last and only question from my side before I will give my approval for this PR: What exactly (which symptom) do the changes in lines149+150 do fix actually? I am okay with every other change proposed in this PR, but don't understand this one. Thx.